server/src/main/java/io/crate/execution/engine/aggregation/impl/NumericStandardDeviationAggregation.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public BoundSignature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public V iterate(RamAccounting ramAccounting,
+                     MemoryManager memoryManager,
+                     V state,
+                     Input<?>... args) throws CircuitBreakingException {
+        if (state != null) {
+            BigDecimal value = (BigDecimal) args[0].value();
+            if (value != null) {
+                long sizeBefore = state.size();
+                state.increment(value);
+                long sizeAfter = state.size();
+                ramAccounting.addBytes(sizeBefore - sizeAfter);
+            }
+        }
+        return state;
+    }
+
+    @Override
+    public V reduce(RamAccounting ramAccounting, V state1, V state2) {
+        if (state1 == null) {
+            return state2;
+        }
+        if (state2 == null) {
+            return state1;
+        }
+        state1.merge(state2);
+        return state1;
+    }
+
+    @Override
+    public boolean isRemovableCumulative() {
+        return true;
+    }
+
+    @Override
+    public V removeFromAggregatedState(RamAccounting ramAccounting, V previousAggState, Input<?>[]stateToRemove) {
+        if (previousAggState != null) {
+            BigDecimal value = (BigDecimal) stateToRemove[0].value();
+            if (value != null) {
+                long sizeBefore = previousAggState.size();
+                previousAggState.decrement(value);
+                long sizeAfter = previousAggState.size();
+                ramAccounting.addBytes(sizeBefore - sizeAfter);
+            }
+        }
+        return previousAggState;
+    }
+
+    @Override
+    public BigDecimal terminatePartial(RamAccounting ramAccounting, V state) {
+        return state.result();
+    }
+
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver referenceResolver,
+                                                       List<Reference> aggregationReferences,
+                                                       DocTableInfo table,
+                                                       Version shardCreatedVersion,
+                                                       List<Literal<?>> optionalParams) {
+        Reference reference = aggregationReferences.get(0);
+        if (reference == null) {
+            return null;
+        }
+        if (!reference.hasDocValues()) {
+            return null;
+        }
+
+        NumericType numericType = (NumericType) reference.valueType();
+        Integer precision = numericType.numericPrecision();
+        Integer scale = numericType.scale();
+        if (precision == null || scale == null) {
+            throw new UnsupportedOperationException(
+                    "NUMERIC type requires precision and scale to support aggregation");
```

Comment:
Does this hold true also for non-docvalues aggregation implementations?

Reply:
nope, for in-flight values:
```
cr> select stddev_samp(a) from unnest([1.0::numeric, 2.0::numeric, 3.0::numeric, 4.0::numeric, 5.0::numeric]) as a;
+--------------------+
|        stddev_samp |
+--------------------+
| 1.5811388300841898 |
+--------------------+
```
but for a column, it's not permitted anyway to create a numeric column without specying precision/scale